packages/storage-plus/src/keys.rs

```diff
@@ -5,6 +5,33 @@ use crate::de::KeyDeserialize;
 use crate::helpers::namespaces_with_key;
 use crate::Endian;
 
+#[derive(Debug)]
+pub enum Key<'a> {
```

Comment:
Okay.... seems simpler than associated types.

Ah, `[u8; N]` is on the stack and easily passed rather than `Vec<u8>` which is on the heap... and we only have to support up to 8 bytes. Nice idea